src/scanner.c

```diff
@@ -0,0 +1,200 @@
+#include <assert.h>
+#include <stdio.h>
+#include <string.h>
+#include <tree_sitter/parser.h>
+
+#define MAX(a, b) ((a) > (b) ? (a) : (b))
```

Comment:
Could you rewrite all these macros to functions?
like this:
```c
int max(int a, int b) {
    return a > b ? a : b;
}
```

The reason is macro don't have types annotations on parameters.
At the first glance, we don't know the type of parameters and the return value.
Another reason is the preprocessor simply text level expand code, it's very easy to make mistake.
I don't know the advantage of macros here. The code complexity and cognitive overhead of macro function is not necessary.

Reply:
That's fine, I can do that

Reply:
could you also rewrite `MAX`?